usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -142,46 +186,92 @@ def delete_docs_by_unique_key(client: Elasticsearch, key: str, value_list: list,
+        lookup_key (str): name of field in targeted elasticsearch index by which we are looking up docs. The field or
+            sub-field provided MUST be of ``keyword`` type (or ``_id`` meta field)
+        value_list (list): if lookup_key field has any of these values, the document will be returned from the lookup
         index (str): Optional name, alias, or pattern of index this query will target. Looks up via config if not
                      provided
 
-    Returns: None
+    Returns: list of values for the ES_AWARDS_UNIQUE_KEY_FIELD fields in the looked-up documents
     """
     if index is None:
         index = f"{config['query_alias_prefix']}-*"
-    col_to_items_dict = defaultdict(list)
-    for line in id_list:
-        col_to_items_dict[line["col"]].append(line["key"])
-    awards = []
-    for column, values in col_to_items_dict.items():
-        values_generator = chunks(values, 1000)
-        for v in values_generator:
-            body = filter_query(column, v)
-            response = client.search(index=index, body=json.dumps(body), size=config["max_query_size"])
-            if response["hits"]["total"]["value"] != 0:
-                awards += [x["_source"][ES_AWARDS_UNIQUE_KEY_FIELD] for x in response["hits"]["hits"]]
-    return awards
 
+    if not _is_allowed_key_field_type(client, lookup_key, index):
+        msg = (
+            f'Cannot perform lookups in index "{index}" with key field "{lookup_key}" because its type is not one of '
+            f"the allowed field types, or the field was not found in that index."
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    if lookup_chunk_size > 65536:
+        # 65,536 is max number of terms that can be added to an ES terms filter query
+        msg = (
+            f"{lookup_chunk_size} is greater than 65,536, which is the max number of terms that can be added to an ES "
+            f"terms filter query"
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    if lookup_chunk_size > config["max_query_size"]:
+        # Some keys would be left undiscovered if our chunk was cut short by the query only returning a lesser subset
+        msg = (
+            f"{lookup_chunk_size} is greater {config['max_query_size']}, which is the max number of query "
+            f"results returnable from this index. Use a smaller chunk or increase query size for this index."
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    award_key_list = []
+    values_generator = chunks(value_list, lookup_chunk_size)
+    for chunk_of_values in values_generator:
+        # Creates an Elasticsearch query criteria
+        q = ES_Q("terms", **{lookup_key: chunk_of_values})
+        response = Search(using=client, index=index).filter(q)[0 : config["max_query_size"]].execute()
```

Comment:
This syntax is crazy, surprised it works.

Is the response limit necessary? If yes, will the records above `max_query_size` be ignored?

Reply:
yeah, it's a stupid syntax but apparently THE syntax for `elasticsearch-dsl`. You use slicing (which they must have overloaded operators) to tell it what. you want the `"size": num` value to be in the request. And yes, above max_query_size are ignored -- i ran into that problem in testing. Truncates to the default of 10

Reply:
wow, crazy. Thanks for the explanation